src/voting.rs

```diff
@@ -471,8 +471,8 @@ async fn add_abuse_vote(ctx: &Context, voter: User, message: &mut Message) {
 
 // This function handles the vote-interactions and the report interaction and
 // calls the appropriate functions for them (logging stuff in the logs)
-pub async fn handle_vote_interaction(ctx: Context, interaction: Interaction) {
-    if let Interaction::MessageComponent(mut component) = interaction {
+pub async fn handle_vote_interaction(ctx: &Context, interaction: &Interaction) {
+    if let Interaction::MessageComponent(mut component) = interaction.to_owned() {
```

Comment:
Why would you make the arguments references, if you call .to_owned() later anyway, creating an owned object usually clones data, maybe revert this?